libs/pika/synchronization/include/pika/synchronization/barrier.hpp

```diff
@@ -88,22 +130,19 @@ namespace pika {
     // barrier::arrival_token is an unspecified type, such that it meets the
     // Cpp17MoveConstructible (Table 28), Cpp17MoveAssignable (Table 30), and
     // Cpp17Destructible (Table 32) requirements.
-
-    template <typename OnCompletion = detail::empty_oncompletion>
+    template <class Completion = detail::empty_completion>
     class barrier
     {
-    public:
-        PIKA_NON_COPYABLE(barrier);
-
-    private:
-        using mutex_type = pika::concurrency::detail::spinlock;
+        std::ptrdiff_t expected;
+        std::atomic<std::ptrdiff_t> expected_adjustment;
+        std::decay_t<Completion> completion;
+        std::atomic<detail::barrier_phase_t> phase;
+        detail::barrier_algorithm_base base;
 
     public:
-        using arrival_token = bool;
+        using arrival_token = detail::barrier_phase_t;
 
-        // Returns:        The maximum expected count that the implementation
-        //                 supports.
-        static constexpr std::ptrdiff_t(max)() noexcept
+        static constexpr std::ptrdiff_t max() noexcept
```

Comment:
Ah, no this should be kept as the old version (even though it's only a potential problem on Windows). The parentheses were exactly to avoid expansion of the Windows macro `max`.